src/EmbeddingSpMDM.cc

```diff
@@ -823,10 +863,19 @@ GenEmbeddingSpMDMLookup<
                 a->vmovups(dst_addr, out_vreg);
               }
             } else {
-              // fp16 output
+              // fp16/bf16 output
               if (instSet == inst_set_t::avx2) {
                 // round nearest with no exception
-                a->vcvtps2ph(out_vreg.xmm(), out_vreg, 8);
+                if (isfp16out) {
+                  a->vcvtps2ph(out_vreg.xmm(), out_vreg, 8);
+                } else if (isbf16out) {
+                  a->vpsrld(tout_vreg, out_vreg, 16);
+                  a->vpand(tout_vreg, tout_vreg, ones_vreg);
+                  a->vpand(tout_vreg, tout_vreg, bf16_bias_vreg);
+                  a->vpaddd(tout_vreg, tout_vreg, out_vreg);
+                  a->vpsrld(tout_vreg, tout_vreg, 16);
+                  a->vpackusdw(tout_vreg, tout_vreg, tout_vreg);
```

Comment:
we don't write back to `out_vreg` here?

Reply:
I deleted `a->vpermq(out_vreg, tout_vreg, 0xd8);` by mistake.
Thanks for remind. Have added it back.

Reply:
It is not covered by UT?

Reply:
I can test it out in Pytorch after force return avx2 here, https://github.com/pytorch/FBGEMM/blob/main/src/Utils.cc#L302。
After fixed it will all be passed.

For FBGEMM ut, I do not test it out. I will investigate it.